src/components/app.tsx

```diff
@@ -653,12 +796,34 @@ export class App extends React.Component<AppProps, AppState> {
                     <hr/>
                     <button onClick={this.tryReconnectModal}>Try again</button>
                 </ReactModal>
+                <ReactModal className="sensor-dialog-content"
+                            overlayClassName="sensor-dialog-overlay"
+                            contentLabel="Bluetooth connection failed"
+                            isOpen={this.state.bluetoothErrorModal} >
+                    <p>{this.messages["bluetooth_connection_failed"]}</p>
+                    <hr/>
+                    <button onClick={this.closeBluetoothErrorModal}>Ok</button>
+                </ReactModal>
                 <div className="app-content">
                     <div className="app-top-bar">
                         {this.renderDualCollectionCheckBox()}
                         <div className="status-message">
                             {this.state.statusMessage || "\xA0"}
                         </div>
+                        {!this.props.sensorManager ?
+                        <div>
+                            <button className="connect-to-device-button smart-focus-highlight disable-focus-highlight"
+                                    onClick={this.handleWiredClick}
+                                    disabled={wirelessConnected || this.state.collecting}>
+                                {wiredConnected ? "Disconnect Wired" : "Connect to Wired"}
+                            </button>
+                            <button className="connect-to-device-button smart-focus-highlight disable-focus-highlight"
+                                    onClick={this.handleWirelessClick}
+                                    disabled={wiredConnected || this.state.collecting}>
+                                {wirelessConnected ? "Disconnect Wireless" : "Connect to Wireless"}
+                            </button>
+                        </div>
+                        : null}
```

Comment:
By supporting both this new implementation and the previous implementation and examples, there are now a handful of connection buttons that we need to keep straight.  Not a big deal, but something to keep an eye on moving forward.